drivers/ethernet/eth_stm32_hal.c

```diff
@@ -227,11 +227,19 @@ static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 	/* Reset TX complete interrupt semaphore before TX request*/
 	k_sem_reset(&dev_data->tx_int_sem);
 
+	/* Unknown incomplete memory write causes TxCplt callback
+	 * to not be called. Fixed by data sync barriers before & after
+	 * HAL tx start function call
+	 */
```

Comment:
```suggestion
	/* Memory barriers (e.g. __DSB()) should flush Tx descriptors into memory. */
	/* Incomplete writing of Tx descriptors may occur with different compilation levels. */
```

Please update the description.